app/src/main/java/com/x8bit/bitwarden/ui/platform/feature/rootnav/RootNavViewModel.kt

```diff
@@ -68,7 +68,13 @@ class RootNavViewModel @Inject constructor(
 
             userState == null ||
                 !userState.activeAccount.isLoggedIn ||
-                userState.hasPendingAccountAddition -> RootNavState.Auth
+                userState.hasPendingAccountAddition -> {
+                if (authRepository.showWelcomeCarousel) {
+                    RootNavState.AuthWithWelcome
+                } else {
+                    RootNavState.Auth
+                }
```

Comment:
Is there any risk that the conditions that produce a value of `authRepository.showWelcomeCarousel` of `true` could change while the user is within the `RootNavState.AuthWithWelcome` flow itself and then some unrelated update to `userState` triggers a switch here from `RootNavState.AuthWithWelcome` to `RootNavState.Auth`?

Reply:
That is why I was suggesting that we put it in the state.

Reply:
The conditions of `authRepository.showWelcomeCarousel` are that the feature flag be enabled and the user has logged in or created an account on that device. I don't think there's a way for the `hasLoggedInOrCreatedAccount` preference to change, so the only way `authRepository.showWelcomeCarousel` could change would be if the feature flag was disabled while the user was viewing the welcome carousel (even then, I'm not sure the flag would update, given that we're not using a synchronous get with a force refresh). But in that case, navigating away from the welcome carousel sounds like desired behavior, right?

Reply:
Gotcha OK. Sounds good then!